### DISCUSSION TITLE
Formato filtro nome cliente

### DISCUSSION THREAD

**Discussion by Raibaz** [fattureincloud/api#305] [Q&A] [answered]
Buongiorno, sto provando a filtrare la lista dei clienti per nome del cliente, ma non riesco a inviare il parametro q nel formato corretto, ricevo l'errore `Invalid query syntax`.

Il formato che sto provando a usare è `name%20%3D%20%27ELVIS%20LIVES%20S.R.L.%27`, che corrisponde all'urlencode di `name = 'ELVIS LIVES S.R.L.'`.

Dove sbaglio?

**Comment by valmoz** [comment, marked as answer]
Buongiorno @Raibaz 
dai nostri log il parametro **q**  inviato nelle sue richieste risulta essere `name%20%3D%20%27ELVIS%20LIVES%20S.R.L.%27`, esattamente come da lei inserito nella richiesta. Il problema è che i nostri log registrano il risultato già decodificato, quindi avrebbe dovuto essere `name = 'ELVIS LIVES S.R.L.'`.

La nostra ipotesi è che il suo HTTP client effettui già l'urlencode e, poiché anche lei effettua l'urlencode, di fatto ci arrivi un parametro encodato due volte.
In pratica, crediamo che il parametro che noi riceviamo sia `name%2520%253D%2520%2527ELVIS%2520LIVES%2520S.R.L.%2527`.

Suggeriamo di riprovare togliendo uno dei due urlencode e di farci sapere il risultato.

Grazie
Mauro

P.S.: solitamente per questo genere di richieste consigliamo di contattare il supporto clienti, in quanto ci permette di avere accesso diretto al company ID e quindi di fare una ricerca mirata sui log.

**Reply by valmoz** [reply]
Buongiorno @Raibaz è riuscito a risolvere?

Grazie
Mauro

**Reply by Raibaz** [reply]
Sì, grazie.
